### ISSUE TITLE
SOAP problems since 1.0RC1 (.NET) - "error reflecting from 'return'"

### ISSUE THREAD

**Issue by GoogleCodeExporter** [bmor1/bimserver#98]
```
What steps will reproduce the problem?
1. Add a Service Reference in Visual Studio 2008 and higher to BiMserver 1.0RC1 
(use http://localhost:8082/services/soap?wsdl or the like as source)
2. Try to ping() the service (ServiceReference_RC1.ServiceInterfaceClient 
client1 = new ServiceReference_RC1.ServiceInterfaceClient(); string str = 
ping("hello");)
3. Exception is thrown

What is the expected output? What do you see instead?
Expected: Input string as ping answere
In fact:

InnerException (translated English):The top XML element 'return' in 
namespace''refers to the distinct types BiMserver.ServiceReference.sUser [] and 
BiMserver.ServiceReference.checkoutResult. Use XML attributes to specify 
another XML name or namespace for the element or types.

Exception: "error reflecting from 'return'";
InnerException (original German): Das oberste XML-Element 'return' im Namespace 
'' verweist auf die Distinct-Typen 
BiMserver.ServiceReference_NB20100917.sUser[] und 
BiMserver.ServiceReference_NB20100917.checkoutResult. Verwenden Sie 
XML-Attribute, um einen anderen XML-Namen oder Namespace für das Element oder 
die Typen anzugeben.

What version of the product are you using? On what operating system?
1.0RC1, Win7 64Bit, Visual Studio 2010

Additional information:
I tested it with several nightly builds - the latest version I have (before 
1.0RC1) is the build from 9th Sep. 2010, which works fine! Since 1.0RC1 it 
doesn't work (including nightly build from 17th Sep. 2010).

```

Original issue reported on code.google.com by `Bastian....@gmx.de` on 18 Sep 2010 at 3:15

**Comment by GoogleCodeExporter** [comment]
Original comment by `berlo...@gmail.com` on 20 Sep 2010 at 11:08
- Added labels: **Milestone-Release1.0**

**Comment by GoogleCodeExporter** [comment]
Original comment by `berlo...@gmail.com` on 20 Sep 2010 at 11:09

**Comment by GoogleCodeExporter** [comment]
```
The email sent to bkarstaedt:

A few changes have been made in the SOAP interface to make it more
usable. One of the changes is that the WSDL now contains meaningful
parameter names for the invokable methods, but I do not suspect that
is causing the issues.

Unfortunately Microsoft is (as usual) not complying with any standards
with their SOAP implementation. There are a few different ways to
provide SOAP, apparently the current one is not working well with
Microsoft's implementation. You can have a look at the following
webpage for more information on the different ways:
http://www.ibm.com/developerworks/webservices/library/ws-whichwsdl/.

As I do not have access to any .Net development tools I cannot test
different setups, so I would like to ask you to do that :). In the
"Shared" project, there is a class called "ServiceInterface". You
should change the following line:

@SOAPBinding(style=Style.RPC, use=Use.LITERAL,
parameterStyle=ParameterStyle.BARE)

I suspect simply changing ParameterStyle.BARE to
ParameterStyle.WRAPPED will do the trick, but you can also try
changing the 'style' and 'use' parameters. Be sure to restart the
server after making changes, and regenerate you .Net classes based on
the new WSDL file.

We are thinking about providing all possible combinations at the same
time, so client developers can choose their own way, but that won't be
implemented in 1.0 yet.

Hope this helps, if you find a working setup, please let us know, as
we will probably make it the default.
```

Original comment by `rdlsoftware@gmail.com` on 23 Sep 2010 at 8:49

**Comment by GoogleCodeExporter** [comment]
```
Default style has been changed to document, this seems to fix the problem with 
.Net
```

Original comment by `rdlsoftware@gmail.com` on 26 Oct 2010 at 11:44
- Changed state: **Fixed**
